src/formatDate.js

```diff
@@ -8,7 +8,49 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const formattedToDateParts = {}; // day, month, year
+  const values = date.split(fromFormat[3]);
+  let formattedDate = '';
+
+  for (let i = 0; i < 3; i++) {
+    if (fromFormat[i] === 'YYYY' && toFormat[i] === 'YY') {
+      formattedToDateParts[2] = values[i].slice(-2);
+    } else if (fromFormat[i] === 'YY' && toFormat[i] === 'YYYY') {
+      const year = Number(values[i]);
+
+      if (year < 30) {
+        formattedToDateParts[2] = '20' + values[i];
+      } else {
+        formattedToDateParts[2] = '19' + values[i];
+      }
+    } else if (fromFormat[i] === 'YY' || fromFormat[i] === 'YYYY') {
+      formattedToDateParts[2] = values[i];
```

Comment:
This condition is redundant because it is covered by the previous conditions. If the input `fromFormat[i]` is either 'YY' or 'YYYY', the code already handles it in the previous branches.